docs/howto/overriding-templates.txt

```diff
@@ -0,0 +1,88 @@
+====================
+Overriding Templates
+====================
+
+Overriding from the project's templates directory
+=================================================
+
+To override templates from a package in ``django.contrib`` or from a third-party
+package, the preferred method is to create replacement templates in your
+project's templates directory.
+
+Let's say you're trying to override the templates for a third-party package
+called ``blog``, which provides the templates ``blog/post.html`` and
+``blog/list.html``. The relevant settings for your project would look like::
+
+    INSTALLED_APPS = [
+        ...,
+        blog,
+        ...,
+    ]
+
+    TEMPLATES = [
+        {
+            'BACKEND': 'django.template.backends.django.DjangoTemplates',
+            'DIRS': [os.path.join(BASE_DIR, 'templates')],
+            'APP_DIRS': True,
+            ...
+        },
+    ]
+
+The :setting:`TEMPLATES` section here should already exist in your project
+settings, as it is normally created when you create a Django project. The
+setting that needs to be modified is :setting:`DIRS<TEMPLATES-DIRS>`.
+
+These settings assume you have a ``templates`` directory in the root of your
+project. To override the templates for the ``blog`` app, you'll create a folder
+in the ``templates`` directory, and add the correct files to that folder::
+
+    templates/
+        blog/
+            list.html
+            post.html
+
+When the configured template loader looks for the ``blog`` app templates, it
+will look in the ``templates`` directory first, because that's what the settings
+above tell it to do. When the views in the ``blog`` app ask for the
+``blog/post.html`` and ``blog/list.html`` templates, the loader will return your
+project's copy of the templates.
+
+Overriding from an app's template directory
+===========================================
+
+While it is safer and cleaner to override templates from the project-level
+templates directory, it is possible to override templates from within an app's
+templates directory.
+
+First, make sure your template settings are checking inside app directories::
+
+    TEMPLATES = [
+        {
+            ...,
+            'APP_DIRS': True,
+            ...
+        },
+    ]
+
+Similar to the example from the previous section, if the app we're working on is
+called ``myapp``, and the app we're trying to override is named ``blog``, then
+our directory structure will look like::
+
+    myapp/
+        ...
+        templates/
+            blog/
+                list.html
+                post.html
+
+With :setting:`APP_DIRS<TEMPLATES-APP_DIRS>` set to ``True`` and the correct
+folder structure and templates, the template loader will look in the app's
+templates directory, find ``blog/list.html`` and ``blog/post.html``, and return
+those.
+
+.. admonition:: Template loader ordering
+
+    If you have a project templates directory, and an app templates directory,
```

Comment:
If you have app and project templates directories that both contain overridden templates, ...